### PR TITLE
Add cobs/compactconstruct module

### PR DESCRIPTION
**Pull request by leoisl**
Add cobs/compactconstruct module. This subtool is very similar to one previously merged (cobs/classicconstruct: https://github.com/nf-core/modules/pull/4586), but it builds a compact bloom filter instead of a classic one, which is the main innovation of this work. For cobs, see https://github.com/iqbal-lab-org/cobs and https://link.springer.com/chapter/10.1007/978-3-030-32686-9_21

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by leoisl** [comment]
Hello @CarsonJM , I am adding you as a reviewer in this module as the command-line interface for this subtool is almost identical to the interface of the `cobs/classicconstruct` subtool, which you just reviewed yesterday (https://github.com/nf-core/modules/pull/4586). Hope that is ok! Thanks a lot again for your help!

**Comment by CarsonJM** [comment]
Not a problem, I'm happy to take a look. I'll get you a review soon!

File: modules/nf-core/cobs/compactconstruct/meta.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - COBS
+  - index
+  - k-mer index
+  - bloom filter
+tools:
+  - "cobs":
+      description: "Compact Bit-Sliced Signature Index (for Genomic k-Mer Data or q-Grams)"
+      homepage: "https://panthema.net/cobs"
+      documentation: "https://github.com/iqbal-lab-org/cobs"
+      tool_dev_url: "https://github.com/iqbal-lab-org/cobs"
```
Comment (mashehu) [inline modules/nf-core/cobs/compactconstruct/meta.yml:15]:
```suggestion
```
It's fine if it is already given as documentation URL 🙂

File: modules/nf-core/cobs/compactconstruct/meta.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      doi: "10.1007/978-3-030-32686-9_21"
+      licence: ["MIT"]
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
```
Comment (mashehu) [inline modules/nf-core/cobs/compactconstruct/meta.yml:24]: please double check if this is actually a plausible example for the meta map.

File: modules/nf-core/cobs/compactconstruct/meta.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  - input:
+      type: file
+      description: |
+        The file or directory to be indexed.
+        COBS can read:
+          1. FASTA files (*.fa, *.fasta, *.fna, *.ffn, *.faa, *.frn, *.fa.gz, *.fasta.gz, *.fna.gz, *.ffn.gz, *.faa.gz, *.frn.gz),
+          2. FASTQ files (*.fq, *.fastq, *.fq.gz., *.fastq.gz),
+          3. "Multi-FASTA" and "Multi-FASTQ" files (*.mfasta, *.mfastq),
+          4. McCortex files (*.ctx),
+          5. or text files (*.txt).
```
Comment (mashehu) [inline modules/nf-core/cobs/compactconstruct/meta.yml:35]:
```suggestion
          1. FASTA files (`*.fa`, `*.fasta`, `*.fna`, `*.ffn`, `*.faa`, `*.frn`, `*.fa.gz`, `*.fasta.gz`, `*.fna.gz`, `*.ffn.gz`, `*.faa.gz`, `*.frn.gz`),
          2. FASTQ files (`*.fq`, `*.fastq`, `*.fq.gz.`, `*.fastq.gz`),
          3. "Multi-FASTA" and "Multi-FASTQ" files (`*.mfasta`, `*.mfastq`),
          4. McCortex files (`*.ctx`),
          5. or text files (`*.txt`).
```

File: modules/nf-core/cobs/compactconstruct/meta.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        You can either recursively scan a directory for all files matching any of these files,
+        or pass a *.list file which lists all paths COBS should index.
```
Comment (mashehu) [inline modules/nf-core/cobs/compactconstruct/meta.yml:37]:
```suggestion
        or pass a `*.list` file which lists all paths COBS should index.
```

**Review by mashehu — APPROVED** [review]

**Review by CarsonJM — APPROVED** [review]
LGTM once all tests pass, nice job on another module!

File: modules/nf-core/cobs/compactconstruct/meta.yml (hunk 2 above)
Comment (leoisl) [reply modules/nf-core/cobs/compactconstruct/meta.yml]: Thanks for bringing this to my attention, I'd always leave the default code created by `nf-core modules create` and wouldn't adapt it to the specific module. Fixed in https://github.com/nf-core/modules/pull/4607/commits/eedde7be21717dd51f0b6021831fb74bbca7ed51 and will also be fixed in my previous modules!

**Comment by leoisl** [comment]
Thanks for your suggestions @mashehu , I accepted and applied all of them!

**Pull request merged**